### PR TITLE
Chore(ci): Add version of format-checks that can run on forks

### PR DESCRIPTION
**Pull request by aaronsteers**
~~Note: this PR is deprioritized in favor of:~~

Brought back as preferred over:

- #37358 

Since the new version is so much faster than the old version, I make a case in the above that we should just _always_ run the unprivileged version.

image

-----

This just adds a simple alternate execution of the `format-check` job. The alternate version will run on forks. The existing version, which requires 8-10 distinct secrets, will be skipped on forks.

For security/consistency reasons, the new check does not use airbyte-ci from the community member's local repo - instead installing from our own master branch (ignoring `airbyte-ci` from the forked repo).

### CODE REVIEW

File: .github/workflows/format_check.yml [hunk 1]
```diff
@@ -87,3 +91,36 @@ jobs:
...
             {\"type\":\"divider\"}]}
+
+  format-check-from-forks:
+    # Same-named job as above, in order to ensure 'required checks' pass either way.
+    # This should run all the same checks as above, except not requiring any credentials.
+    # This may be slower than the version that runs with creds, due to the affect of
+    # caching.
+    name: "Check for formatting errors"
+    if: >
+      true ||
```
Comment (aaronsteers) [inline .github/workflows/format_check.yml:102]: To-do before merging:


```suggestion
```

File: .github/workflows/format_check.yml [hunk 2]
```diff
@@ -12,6 +12,10 @@ jobs:
   format-check:
     # IMPORTANT: This name must match the require check name on the branch protection settings
     name: "Check for formatting errors"
+    if: >
+      false &&
```
Comment (aaronsteers) [inline .github/workflows/format_check.yml:16]:
```suggestion
```

**Comment by aaronsteers** [comment]
Closing in favor of 

- #37358 

We can re-open this if we decide to go a different direction.

File: .github/workflows/format_fix.yml [hunk 3]
```diff
@@ -20,6 +23,7 @@ jobs:
           # Important that this is set so that CI checks are triggered again
           # Without this we would be forever waiting on required checks to pass
           token: ${{ secrets.GH_PAT_APPROVINGTON_OCTAVIA }}
+          fetch-depth: 0
```
Comment (aaronsteers) [inline .github/workflows/format_fix.yml:26]:
```suggestion
          fetch-depth: 1
```

File: .github/workflows/format_fix.yml [hunk 4]
```diff
@@ -12,6 +12,9 @@ jobs:
   format-fix:
     name: "Run airbyte-ci format fix all"
     runs-on: ubuntu-latest
+    permissions:
+      # Allow writing to repository
+      contents: write
```
Comment (aaronsteers) [inline .github/workflows/format_fix.yml:17]:
```suggestion
```

File: .github/workflows/format_fix.yml [hunk 5]
```diff
@@ -20,6 +23,7 @@ jobs:
           # Important that this is set so that CI checks are triggered again
           # Without this we would be forever waiting on required checks to pass
           token: ${{ secrets.GH_PAT_APPROVINGTON_OCTAVIA }}
+          fetch-depth: 1
```
Comment (aaronsteers) [inline .github/workflows/format_fix.yml:26]:
```suggestion
```

File: .github/workflows/format_check.yml [hunk 6]
```diff
@@ -87,3 +90,35 @@ jobs:
...
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install `airbyte-ci` CLI
+        # We use the stable `airbyte-ci` from `master`. Ignores any CI changes
+        # local to this repo.
+        run: >
+          pip install
```
Comment (aaronsteers) [inline .github/workflows/format_check.yml:119]:
```suggestion
     - name: Install Airbyte CI from binary 
       id: install-airbyte-ci-binary 
       shell: bash 
       run: | 
         curl -sSL "https://connectors.airbyte.com/airbyte-ci/releases/ubuntu/latest/airbyte-ci" --output airbyte-ci-bin 
         sudo mv airbyte-ci-bin /usr/local/bin/airbyte-ci 
         sudo chmod +x /usr/local/bin/airbyte-ci
```

Binary install from https://github.com/airbytehq/airbyte/blob/dfe0b74a4d6f5cfc15a910cfb4c89821f7d2394c/.github/actions/install-airbyte-ci/action.yml#L35-L42

File: .github/workflows/format_check.yml [hunk 7]
```diff
@@ -87,3 +90,35 @@ jobs:
...
+          git+https://github.com/airbytehq/airbyte.git#subdirectory=airbyte-ci/connectors/pipelines
+
+      - name: Run format checks
+        run: |
+          airbyte-ci format check all
```
Comment (aaronsteers) [inline .github/workflows/format_check.yml:124]:
```suggestion
          /usr/local/bin/airbyte-ci format check all
```

File: .github/workflows/format_check.yml [hunk 8]
```diff
@@ -87,3 +90,37 @@ jobs:
...
             {\"type\":\"divider\"}]}
+
+  format-check-from-forks:
+    # Same-named job as above, in order to ensure 'required checks' pass either way.
+    # This should run all the same checks as above, except not requiring any credentials.
+    # This may be slower than the version that runs with creds, due to the affect of
+    # caching.
+    name: "Check for formatting errors"
+    if: >
+      github.event.pull_request.head.repo.fork == true
```
Comment (aaronsteers) [inline .github/workflows/format_check.yml:101]:
```suggestion
    # TODO: Revert the 'true' part
    if: >
      true ||
      github.event.pull_request.head.repo.fork == true
```

File: .github/workflows/format_check.yml [hunk 9]
```diff
@@ -87,3 +90,39 @@ jobs:
...
+  format-check-from-forks:
+    # Same-named job as above, in order to ensure 'required checks' pass either way.
+    # This should run all the same checks as above, except not requiring any credentials.
+    # This may be slower than the version that runs with creds, due to the affect of
+    # caching.
+    name: "Check for formatting errors"
+    # TODO: Revert the 'true' part
+    if: >
+      true ||
+      github.event.pull_request.head.repo.fork == true
```
Comment (aaronsteers) [inline .github/workflows/format_check.yml:103]:
```suggestion
      true ||
      (github.event.pull_request.head.repo.fork == true)
```

File: .github/workflows/format_check.yml [hunk 10]
```diff
@@ -87,3 +90,39 @@ jobs:
...
+
+      - name: Install Airbyte CI from binary 
+        id: install-airbyte-ci-binary 
+        shell: bash 
+        run: | 
+          curl -sSL "https://connectors.airbyte.com/airbyte-ci/releases/ubuntu/latest/airbyte-ci" --output airbyte-ci-bin 
+          sudo mv airbyte-ci-bin /usr/local/bin/airbyte-ci 
+          sudo chmod +x /usr/local/bin/airbyte-ci
+          git+https://github.com/airbytehq/airbyte.git#subdirectory=airbyte-ci/connectors/pipelines
+
```
Comment (aaronsteers) [inline .github/workflows/format_check.yml:125]:
```suggestion

```

File: .github/workflows/format_check.yml [hunk 11]
```diff
@@ -87,3 +90,37 @@ jobs:
...
+  format-check-from-forks:
+    # Same-named job as above, in order to ensure 'required checks' pass either way.
+    # This should run all the same checks as above, except not requiring any credentials.
+    # This may be slower than the version that runs with creds, due to the affect of
+    # caching.
+    name: "Check for formatting errors"
+    # TODO: Revert the 'true' part
+    if: >
+      true ||
+      (github.event.pull_request.head.repo.fork == true)
```
Comment (aaronsteers) [inline .github/workflows/format_check.yml:103]:
```suggestion
  format-check-from-forks:
    # Same-named job as above, in order to ensure 'required checks' pass either way.
    # This should run all the same checks as above, except not requiring any credentials.
    name: "Check for formatting errors"
    if: >
      github.event.pull_request.head.repo.fork == true
```

**Review by natikgadzhi — APPROVED** [review]
LGTM with caveat that Augustin then cleans up the "consistency" piece when run-airbyte-ci is ready to run with optional tokens. Let's merge this!

**Pull request merged**